src/gui/instrument/InstrumentTrackWindow.cpp

```diff
@@ -710,5 +698,75 @@ void InstrumentTrackWindow::adjustTabSize(QWidget *w)
 	w->update();
 }
 
+QMdiSubWindow* InstrumentTrackWindow::findSubWindowInParents()
+{
+	// TODO Move to helper? Does not seem to be provided by Qt.
+	auto p = parentWidget();
+
+	while (p != nullptr)
+	{
+		auto mdiSubWindow = dynamic_cast<QMdiSubWindow*>(p);
+		if (mdiSubWindow)
+		{
+			return mdiSubWindow;
+		}
+		else
+		{
+			p = p->parentWidget();
+		}
+	}
+
+	return nullptr;
+}
+
+void InstrumentTrackWindow::updateSubWindow()
+{
+	auto subWindow = findSubWindowInParents();
+	if (subWindow && m_instrumentView)
+	{
+		Qt::WindowFlags flags = subWindow->windowFlags();
+
+		const auto instrumentViewResizable = m_instrumentView->isResizable();
+
+		if (instrumentViewResizable)
+		{
+			// TODO As of writing SlicerT is the only resizable instrument. Is this code specific to SlicerT?
+			const auto extraSpace = QSize(12, 208);
+			subWindow->setMaximumSize(m_instrumentView->maximumSize() + extraSpace);
+			subWindow->setMinimumSize(m_instrumentView->minimumSize() + extraSpace);
+
+			flags &= ~Qt::MSWindowsFixedSizeDialogHint;
```

Comment:
Consider also adding `setFixedSize()` here. `MSWindowsFixedSizeDialogHint` annoyingly appears to be a windows-only thing, or at least, isn't respected on wayland, while `setFixedSize()` or `layout->setSizeConstraint(QLayout::SetFixedSize` is.

Reply:
@SpomJ , do you mean to set the `subWindow` to a fixed size? If yes, then I am not sure if that makes sense. The calls to `setMaximumSize` and `setMinimumSize` make the sub window resizable according to the limits as defined by the instrument view. Setting a fixed size would take that away. And there would also be the question which fixed size to set?

Reply:
I agree that this is counter productive. But anyways - this PR is about maximizing, not about min/max size (though the diff might look like it at first sight) - min/max size is a job which #7524.

Reply:
Resolving this conversation then. @SpomJ, feel free to reopen it if necessary.